From 861ba3143001caf2623ce5d84a1d04a69b502339 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20G=C3=B3rny?= <mgorny@gentoo.org>
Date: Mon, 11 May 2020 08:52:04 +0200
Subject: [PATCH 1/2] Support overriding CC, respect *FLAGS in fake-curl

The shipped libraries (obviously) do not work on non-amd64 platforms,
and rebuilding without the correct compiler and flags does not work
on more complex setups such as ppc64 with 32-bit userland.  Make
the Makefile permit CC override while preserving the current default,
and use user-provided CFLAGS, CPPFLAGS and LDFLAGS.
---
 tests/fake-curl/libcurl/Makefile | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/fake-curl/libcurl/Makefile b/tests/fake-curl/libcurl/Makefile
index b05bc89..8ece206 100644
--- a/tests/fake-curl/libcurl/Makefile
+++ b/tests/fake-curl/libcurl/Makefile
@@ -7,8 +7,11 @@ all: \
 
 .SUFFIXES: .c .so
 
+CC = `curl-config --cc`
+CFLAGS += `curl-config --cflags`
+
 .c.so:
-	`curl-config --cc` `curl-config --cflags` -shared -fPIC \
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -shared -fPIC \
 		-Wl,-soname,$@ -o $@ $<
 
 show-targets:
-- 
2.26.2

